Source/Core/AudioCommon/Mixer.h

```diff
@@ -72,26 +159,35 @@ class Mixer final
     }
     void DoState(PointerWrap& p);
     void PushSamples(const short* samples, unsigned int num_samples);
-    unsigned int Mix(short* samples, unsigned int numSamples, bool consider_framelimit,
-                     float emulationspeed, int timing_variance);
+    void Mix(short* samples, unsigned int num_samples);
     void SetInputSampleRateDivisor(unsigned int rate_divisor);
     unsigned int GetInputSampleRateDivisor() const;
     void SetVolume(unsigned int lvolume, unsigned int rvolume);
     std::pair<s32, s32> GetVolume() const;
-    unsigned int AvailableSamples() const;
 
   private:
     Mixer* m_mixer;
     unsigned m_input_sample_rate_divisor;
     bool m_little_endian;
-    std::array<short, MAX_SAMPLES * 2> m_buffer{};
-    std::atomic<u32> m_indexW{0};
-    std::atomic<u32> m_indexR{0};
+
+    size_t m_buffer_index = 0;
+    GranuleBuffer m_buffer{};
+
+    uint32_t m_current_index;
+    Granule m_front, m_back;
+
+    std::array<Granule, GRANULE_QUEUE_SIZE> m_queue;
+    std::atomic<size_t> m_queue_head{0};
+    std::atomic<size_t> m_queue_tail{0};
+    std::atomic<bool> m_queue_looping{false};
+    size_t m_queue_fade_index = 0;
+
+    void enqueue(const Granule& granule);
+    void dequeue(Granule& granule);
```

Comment:
`Dequeue` and also pass a pointer to `Granule` so it's more obviously a modified value at the call-site, per style-guide.